docs/source/API/core/execution_spaces.rst

```diff
@@ -310,6 +311,27 @@ Typedefs
 
 * ``scratch_memory_space``: The ``ScratchMemorySpace`` that parallel patterns will use for allocation of scratch memory (for instance, as requested by a |KokkosTeamPolicy|_)
 
+* ``size_type``: The default integer type associated with this space. Used as preferred type for indexing.
+
+.. list-table::
+ :widths: 25 25
+ :header-rows: 1
+
+ * - Space
+   - ``size_type``
+ * - ``Cuda``
+   - ``unsigned int``
+ * - ``HIP``
+   - ``unsigned int``
+ * - ``SYCL``
+   - ``int``
+ * - ``OpenACC``
+   - ``size_t``
+ * - ``OpenMPTarget``
+   - ``unsigned int``
+ * - All host spaces
+   - ``size_t``
```

Comment:
I am not sure whether I would want to spell out what these currently are.
Maybe instead we could comment that they are implementation defined and that backends effectively have picked different signed or unsigned integer types with width 32 or 64 bits.

Reply:
I agree.

Reply:
Sounds good

Reply:
done